tests/distributions/test_distribution.py

```diff
@@ -728,6 +728,34 @@ def inner_dist(size=None):
             pm.logp(pm.LogNormal.dist(), 1.0).eval(),
         )
 
+    def test_signature(self):
+        def dist(p, size):
+            return -pm.Categorical.dist(p=p, size=size)
+
+        out = CustomDist.dist([0.25, 0.75], dist=dist, signature="(p)->()")
+        # Size and updates are added automatically to the signature
+        assert out.owner.op.signature == "(),(p)->(),()"
+        assert out.owner.op.ndim_supp == 0
+        assert out.owner.op.ndims_params == [0, 1]
+
+        # When recreated internally, the whole signature may already be known
+        out = CustomDist.dist([0.25, 0.75], dist=dist, signature="(),(p)->(),()")
+        assert out.owner.op.signature == "(),(p)->(),()"
+        assert out.owner.op.ndim_supp == 0
+        assert out.owner.op.ndims_params == [0, 1]
+
+        # A safe signature can be inferred from ndim_supp and ndims_params
+        out = CustomDist.dist([0.25, 0.75], dist=dist, ndim_supp=0, ndims_params=[0, 1])
+        assert out.owner.op.signature == "(),(i10)->(),()"
+        assert out.owner.op.ndim_supp == 0
+        assert out.owner.op.ndims_params == [0, 1]
+
+        # Otherwise be default we assume everything is scalar, even though it's wrong in this case
```

Comment:
Will having a wrong signature cause an error somewhere down the pipe we can check with an assert_raises here?

I also prefer to split these tests cases into individual functions to make it more obvious what isn't working when a test fails.

Reply:
Pytest shows which line failed, this is more readable than parametrize imo. 

Parametrize is more readable when you have a loop in the test because the line doesn't tell in which iteration of the loop it failed.

Reply:
Resolving without changes